core/vtk/ttkMergeTreeDistanceMatrix/ttkMergeTreeDistanceMatrix.h

```diff
@@ -0,0 +1,197 @@
+/// \ingroup vtk
+/// \class ttkMergeTreeDistanceMatrix
+/// \author Mathieu Pont <mathieu.pont@lip6.fr>
+/// \date 2021.
+///
+/// \brief TTK VTK-filter that wraps the ttk::MergeTreeDistanceMatrix module.
+///
+/// This VTK filter uses the ttk::MergeTreeDistanceMatrix module to compute the
+/// distance matrix of a group of merge trees.
+///
+/// \param Input vtkMultiBlockDataset
+/// \param Output vtkTable
+///
+/// This filter can be used as any other VTK filter (for instance, by using the
+/// sequence of calls SetInputData(), Update(), GetOutputDataObject()).
+///
+/// See the related ParaView example state files for usage examples within a
+/// VTK pipeline.
+///
+/// \sa ttk::MergeTreeDistanceMatrix
+/// \sa ttkAlgorithm
+
+#ifndef _TTKMERGETREEDISTANCEMATRIX_
+#define _TTKMERGETREEDISTANCEMATRIX_
+
+#pragma once
+
+// VTK Module
+#include <ttkMergeTreeDistanceMatrixModule.h>
+
+// VTK Includes
+#include <ttkAlgorithm.h>
+#include <vtkMultiBlockDataSet.h>
+
+// TTK Base Includes
+#include <MergeTreeDistanceMatrix.h>
+
+class TTKMERGETREEDISTANCEMATRIX_EXPORT ttkMergeTreeDistanceMatrix
+  : public ttkAlgorithm // we inherit from the generic ttkAlgorithm class
+  ,
+    protected ttk::MergeTreeDistanceMatrix // and we inherit from the base class
+{
+private:
+  /**
+   * Add all filter parameters only as private member variables and
+   * initialize them here.
+   */
+  // Execution Options
+  int Backend = 0;
+
+public:
+  /**
+   * Automatically generate getters and setters of filter
+   * parameters via vtkMacros.
+   */
+  // Input Options
+  void SetEpsilon1UseFarthestSaddle(bool epsilon1UseFarthestSaddle) {
+    epsilon1UseFarthestSaddle_ = epsilon1UseFarthestSaddle;
+    Modified();
+  }
+  bool GetEpsilon1UseFarthestSaddle() {
+    return epsilon1UseFarthestSaddle_;
+  }
+
+  void SetEpsilonTree1(double epsilonTree1) {
+    epsilonTree1_ = epsilonTree1;
```

Comment:
actually, we recommend to use our variable naming convention (camel wording, starting with a small letter and finishing with `_`) only for *base* classes.
for vtk classes, we recommend to follow the vtk convention (camel wording, starting with a capital).
in that example, if your variable's name followed the vtk convetion, you could declare getters and setters with the standard vtk macros `vtkSetMacro()` and `vtkGetMacro()`.
see other examples throughout the code to see how to use them.
you may need to address this in your prior PRs too (#633 , #638 )

Reply:
Actually `epsilonTree1_` is a variable of the `MergeTreeBase` class and `ttkMergeTreeDistanceMatrix` inherits from this class through `MergeTreeDistanceMatrix`.
Therefore its indeed a variable of the base layer and hence the definition of the setter/getter without vtk macros.
I did like this to avoid redefining these parameters (`epsilonTree1_` etc.) in the vtk layer.
Should we do differently here?

Reply:
I see, that makes sense and this is compatible with other instances (such as `ttkAlgorithm::SetDebugLevel()`).